llvm/test/tools/sycl-post-link/sycl-sanitize/asan.ll

```diff
@@ -0,0 +1,66 @@
+; This test checks that the post-link tool properly generates "asanUsed=1"
+; in [SYCL/misc properties]
+
+; RUN: sycl-post-link -split=kernel -symbols -S < %s -o %t.table
+; RUN: FileCheck %s -input-file=%t_0.prop
+; CHECK: [SYCL/misc properties]
+; CHECK: asanUsed=1
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+%struct.AssertHappened.10 = type { i32, [257 x i8], [257 x i8], [129 x i8], i32, i64, i64, i64, i64, i64, i64 }
+%"class.sycl::_V1::range.12" = type { %"class.sycl::_V1::detail::array.11" }
+%"class.sycl::_V1::detail::array.11" = type { [2 x i64] }
+%"class.sycl::_V1::detail::RoundedRangeIDGenerator.15" = type <{ %"class.sycl::_V1::range.12", %"class.sycl::_V1::range.12", %"class.sycl::_V1::range.12", %"class.sycl::_V1::range.12", i8, [7 x i8] }>
+%"class.sycl::_V1::detail::RoundedRangeKernel.17" = type <{ %"class.sycl::_V1::range.12", %class.anon.16, [7 x i8] }>
+%class.anon.16 = type { i8 }
+%struct.__devicelib_div_t_32.13 = type { i32, i32 }
+%struct.__devicelib_div_t_64.14 = type { i64, i64 }
+
+@llvm.global_ctors = appending global [1 x { i32, ptr, ptr }] [{ i32, ptr, ptr } { i32 1, ptr @asan.module_ctor, ptr null }]
+@__spirv_BuiltInGlobalSize = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+@__spirv_BuiltInGlobalInvocationId = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+@__spirv_BuiltInLocalInvocationId = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+@SPIR_AssertHappenedMem = linkonce_odr dso_local addrspace(1) global %struct.AssertHappened.10 zeroinitializer, align 8
+@__spirv_BuiltInWorkgroupId = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+@__spirv_BuiltInGlobalLinearId = external dso_local local_unnamed_addr addrspace(1) constant i64, align 8
+@__spirv_BuiltInWorkgroupSize = external dso_local local_unnamed_addr addrspace(1) constant <3 x i64>, align 32
+
+; Function Attrs: nounwind uwtable
+define internal void @asan.module_ctor() #0 {
+  call void @__asan_init()
+  call void @__asan_version_mismatch_check_v8()
+  ret void
+}
+
+declare void @__asan_init()
```

Comment:
this lit could be simplified to contain this line only?

Reply:
Nope, if so, the generated prop file will be incorrect.